wave_lang/kernel/wave/wave.py

```diff
@@ -567,18 +569,64 @@ def compile_to_mlir(
             llvm_func_config,
         )
 
-        emitter = WaveEmitter(
-            dispatch_entrypoint, trace, self.constraints, options, self.grid_type
-        )
-        try:
-            emitter.emit(trace.get_root_graph())
-        except:
-            logger.info("Error in emitter")
-            asm = mb.module_op.get_asm()
-            logger.info(asm)
-            raise
-        emitter.finish()
+        print("AFTER ", mb.module_op.get_asm())
+        if not module_op:
+            emitter = WaveEmitter(
+                dispatch_entrypoint, trace, self.constraints, options, self.grid_type
+            )
+            try:
+                emitter.emit(trace.get_root_graph())
+            except:
+                logger.info("Error in emitter")
+                asm = mb.module_op.get_asm()
+                logger.info(asm)
+                raise
+            emitter.finish()
+        else:
+            with exe._loc, InsertionPoint.at_block_begin(dispatch_entrypoint.entry_block):
+                target_block = dispatch_entrypoint.entry_block
+                source_func_op = list(module_op.operation.regions[0].blocks[0])[0]
+                source_block = source_func_op.regions[0].blocks[0]
+
+                source_args = list(source_block.arguments)
+                target_args = list(target_block.arguments)
+
+                def convert_memref_to_stream_binding(target_block, old_arg, new_arg, index):
+                    """Convert a memref argument to stream.binding + subspan extraction."""
+                    # Create zero constant
+                    result_type = IndexType.get()
+                    zero_value = arith_d.constant(result_type, IntegerAttr.get(result_type, 0))
+
+                    # Create subspan operation
+                    subspan_op = stream_d.binding_subspan(
+                        old_arg.type,  # The original memref type
+                        new_arg,       # The stream.binding argument
+                        byte_offset=zero_value,
+                      # dynamic_dims=dispatch_entrypoint.get_dynamic_dims(binding),
+                        dynamic_dims=[], # TODO: get dynamic dims
```

Comment:
Can we error out when dynamic dims are present instead of silently dropping them?